src/server/ua_services_session.c

```diff
@@ -72,6 +154,42 @@ Service_ActivateSession(UA_Server *server, UA_SecureChannel *channel,
         return;
     }
 
+    if(channel->securityMode == UA_MESSAGESECURITYMODE_SIGN ||
```

Comment:
As long as i understood the standard correctly, this is how it should be. But i'm not sure about this.